Sources/Diff/CollectionViewManager+Diff.swift

```diff
@@ -36,10 +63,69 @@ extension CollectionViewManager {
         }
     }
 
+    // MARK: - Private
+
+    private func logBadItems(for sectionItems: [CollectionViewDiffSectionItem]) {
+        var badSectionItems: [CollectionViewSectionItem] = []
+        var badCellItems: [CollectionViewCellItem] = []
+        var badReusableViewItems: [CollectionViewReusableViewItem] = []
+
+        for sectionItem in sectionItems {
+            if sectionItem.diffIdentifier.isEmpty {
+                badSectionItems.append(sectionItem)
+            }
+            for cellItem in sectionItem.cellItems {
+                guard let diffCellItem = cellItem as? CollectionViewDiffCellItem else {
+                    badCellItems.append(cellItem)
+                    continue
+                }
+                if diffCellItem.diffIdentifier.isEmpty {
+                    badCellItems.append(diffCellItem)
+                }
+            }
+            for viewItem in sectionItem.reusableViewItems {
+                guard let diffViewItem = viewItem as? CollectionViewDiffReusableViewItem else {
+                    badReusableViewItems.append(viewItem)
+                    continue
+                }
+                if diffViewItem.diffIdentifier.isEmpty {
+                    badReusableViewItems.append(diffViewItem)
+                }
+            }
+        }
+        if badSectionItems.isEmpty,
+           badCellItems.isEmpty,
+           badReusableViewItems.isEmpty {
+            return
+        }
+
+        print("⚠️ CollectionViewTools Warning!!! Following items have empty diffIdentifier:")
+        for sectionItem in badSectionItems {
+            if let index = sectionItem.index {
+                print("\(type(of: sectionItem)) at index \(index)")
+            }
+        }
+        for viewItem in badReusableViewItems {
+            if let index = viewItem.sectionItem?.index {
+                print("\(type(of: viewItem)) in section at index \(index)")
+            }
+        }
+        for cellItem in badCellItems {
+            if let indexPath = cellItem.indexPath {
+                print("\(type(of: cellItem)) at indexpath (\(indexPath.section), \(indexPath.item))")
+            }
+        }
+    }
+    
     private func updateWithoutAnimation(sectionItems: [CollectionViewDiffSectionItem], shouldReload: Bool) {
         _sectionItems = sectionItems
         registerSectionItems()
         recalculateIndexes()
+
+        #if DEBUG
+        logBadItems(for: sectionItems)
+        #endif
```

Comment:
This functionality can be very helpful during the development, but if I'm not mistaken - this code won't work with Carthage-builded frameworks. Maybe we can move this to some kind of CollectionViewManager settings? 🤔

Reply:
Good point. What do you think if I just remove #if DEBUG at all?

Reply:
I think doing `logBadItems` in production is bad because of pretty big overhead it will introduce.
Checking for an environment variable would make the most sense here, IMO.